### ISSUE TITLE
Vercel Deployment Not Loading Swiper CDN

### ISSUE THREAD

**Issue by nardis9501** [nardis9501/Famdesc-landing-blog#10]
### Description:
When deploying to Vercel, the Swiper CDN is not being executed, causing the Swiper component not to function properly on the deployed site. This issue is not present in the local development environment.

### Steps to Reproduce (if applicable):
1. Deploy the project to Vercel.
2. Navigate to the page with the Swiper component.
3. Observe that the Swiper component is not functioning as expected.

### Expected Behavior:
The Swiper CDN should be executed, and the Swiper component should function properly in the deployed environment.

### Actual Behavior:
The Swiper CDN is not being executed, resulting in the Swiper component not functioning in the deployed environment on Vercel.

### Screenshots or Logs (if applicable):
![Swiper Component Not Working](https://github.com/user-attachments/assets/eba48a4a-51d5-4475-b5ea-ea54436ad681)

### Environment (if applicable):
- OS: N/A (Issue is with Vercel deployment)
- Browser: N/A (Occurs across all browsers)
- Version: N/A

### Additional Context:
The current Content-Security-Policy in the Vercel configuration may be blocking the Swiper CDN from being executed. Here is the relevant part of the configuration:

```json
{
  "key": "Content-Security-Policy",
  "value": "default-src 'self'; base-uri 'self'; form-action 'self'; frame-src 'self'; frame-ancestors 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src 'self' data: https://images.unsplash.com; connect-src 'self'; object-src 'none'; upgrade-insecure-requests; block-all-mixed-content"
}

**Comment by nardis9501** [comment]
### Issue Explanation

The issue was resolved by adjusting the HTTP headers configuration in the `vercel.json` file. These headers are crucial for handling CORS (Cross-Origin Resource Sharing) and enhancing the security and performance of your website. Below is an explanation of the importance of each configured header:

**Content-Security-Policy (CSP):**
This header defines restrictions on the resources that can be loaded on your page. In this case, it allows resources to be loaded only from the same domain (`'self'`) and from `https://images.unsplash.com`. It also allows inline scripts (`'unsafe-inline'`) and script evaluation (`'unsafe-eval'`). It's important to note that the initial CSP was restricting the execution of scripts from our CDN, preventing Swiper.js from working correctly. By adjusting this policy, we allowed the necessary scripts to be loaded from trusted external sources.

**Permissions-Policy:**
Controls the browser APIs and features available to your site. In this case, it is empty (`interest-cohort=()`), which should not affect the execution of your CDN.

**Referrer-Policy:**
Defines how the `Referer` header is sent when navigating between pages. Here it is set to `no-referrer-when-downgrade`, meaning the `Referer` will only be sent when the navigation is secure (e.g., from HTTPS to HTTPS).

**X-Content-Type-Options:**
Helps prevent MIME type attacks. It is set to `nosniff`, instructing the browser not to guess the content type based on the file extension.

**X-Frame-Options:**
Protects against clickjacking attacks by controlling whether your site can be displayed within a frame (`iframe`). Here it is set to `SAMEORIGIN`, allowing it to be displayed only on the same domain.

**X-XSS-Protection:**
Helps prevent cross-site scripting (XSS) attacks. It is set to `1; mode=block`, enabling XSS protection in the browser.

**Cache-Control:**
Defines how resources should be cached. Here it is set to `public, max-age=0, must-revalidate`, indicating that resources can be publicly cached but must be revalidated before use.

**Strict-Transport-Security (HSTS):**
Enhances security by enforcing secure (HTTPS) connections. Here it is set to `max-age=31536000; includeSubDomains; preload`, indicating that HSTS should be applied for one year, including subdomains, and be preloaded in browsers.

### Solution

To resolve the issue of Swiper.js not loading correctly in the Vercel deployment, the `Content-Security-Policy` policies in the `vercel.json` file were adjusted to allow script loading from the Swiper CDN. This ensured that the necessary resources could be loaded without undue restrictions, allowing the slider functionality to work correctly.

---

In summary, these headers are essential for the security and performance of your website. Adjusting them appropriately allowed Swiper.js to execute correctly in the production environment.
